static/src/javascripts/projects/common/modules/identity/email-preferences.js

```diff
@@ -0,0 +1,131 @@
+// This is a workaround for the email preferences page https://profile.thegulocal.com/email-prefs
+// We want to submit subscribe/unsubscribe requests without a full page refresh
+// Hopefully this will be short-lived; if it is still alive in 2017, git blame and cry
+
+define([
+    'bean',
+    'bonzo',
+    'qwery',
+    'reqwest',
+    'fastdom',
+    'common/utils/$'
+], function (
+    bean,
+    bonzo,
+    qwery,
+    reqwest,
+    fastdom,
+    $
+) {
+    function reqwestEmailSubscriptionUpdate(buttonEl) {
+        bean.on(buttonEl, 'click', function () {
+            buttonEl.disabled = true;
+            buttonEl.innerHTML = 'Loading...';
+            var formQueryString = generateFormQueryString(buttonEl);
+            reqwest({
+                url: '/email-prefs',
+                method: 'POST',
+                data: formQueryString,
+                error: function () {
+                    renderErrorMessage(buttonEl);
+                },
+                success: function (response) {
+                    var subscriptionState;
+                    if (response && response.subscriptions && response.subscriptions.length) {
+                        subscriptionState = response.subscriptions[0].subscribedTo;
+                    }
+                    try {
+                        if (response.subscriptions.length < 1) {
+                            subscriptionState = false;
+                        } else {
+                            subscriptionState = response.subscriptions[0].subscribedTo;
+                        }
+                    } catch (err) {
+                        renderErrorMessage(buttonEl);
+                    } finally {
+                        updateButton(buttonEl, subscriptionState);
+                    }
+                }
+            });
+        });
+    }
+
+    function enhanceEmailPreferences() {
+        $.forEachElement('.email-subscription__button', reqwestEmailSubscriptionUpdate);
+        $.forEachElement('.save__button', reqwestEmailSubscriptionUpdate);
+    }
+
+    function encodeFormData(csrfToken, buttonVal, htmlPreference) {
+        return 'csrfToken=' + encodeURIComponent(csrfToken) + '&' +
+        'addEmailSubscription=' + encodeURIComponent(buttonVal) + '&' +
+        'htmlPreference=' + encodeURIComponent(htmlPreference);
+    }
+
+    function renderErrorMessage(buttonEl) {
+        return fastdom.write(function () {
+            clearErrorMessages();
+            var errorMessage = bonzo(bonzo.create(
+                '<div class="form__error">' +
+                    'Sorry, an error has occurred, please refresh the page and try again' +
+                '</div>'
+            ));
+            bonzo(errorMessage).insertAfter(buttonEl.parentNode);
+        });
+    }
+
+    function clearErrorMessages() {
+        if (qwery('.form__error')) {
+            $.forEachElement('.form__error', function (errorEl) {
+                errorEl.remove();
```

Comment:
For some crazy reason `.remove()` is not available in IE. I would use `$().remove` (or you can do `errorEl.parentNode.removeChild(errorEl)` but that's a bit ugly).